src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php

```diff
@@ -114,6 +142,8 @@ protected function doGetArguments(Request $request, $controller, array $paramete
                 $arguments[] = $request;
             } elseif ($param->isDefaultValueAvailable()) {
                 $arguments[] = $param->getDefaultValue();
+            } elseif ($this->supportsParameterType && null !== ($type = $param->getType()) && $type->allowsNull()) {
```

Comment:
Yes, but `allowsNull` is also available in PHP 5. It will have the same behavior but with a simpler / more performant code.